src/types/proof/definition.rs

```diff
@@ -152,6 +158,14 @@ impl<H: SimpleHasher> SparseMerkleProof<H> {
             }
         }
 
+        self.unsafe_verify(expected_root_hash, element_key)?;
+
+        Ok(())
+    }
+
+    /// A helper function that is used to verify the values of the Merkle tree against the root hash without
+    /// any preliminary checks
+    fn unsafe_verify(&self, expected_root_hash: RootHash, element_key: KeyHash) -> Result<()> {
```

Comment:
nit: `unsafe` has a specific meaning in rust. Unless this function can cause memory corruption if used improperly, I think `verify_unchecked` or similar is more idiomatic